src/Tmds.Ssh/SftpChannel.cs

```diff
@@ -1079,10 +1085,11 @@ private async ValueTask DownloadFileAsync(string remotePath, string localPath, l
             permissions = attributes.Permissions;
         }
 
-        using FileStream localFile = OpenFileStream(localPath, overwrite ? FileMode.Create : FileMode.CreateNew, FileAccess.Write, FileShare.None, permissions!.Value);
+        using FileStream? localFile = localPath is null ? null : OpenFileStream(localPath, overwrite ? FileMode.Create : FileMode.CreateNew, FileAccess.Write, FileShare.None, permissions.Value);
+        destination ??= localFile;
+        var writeSync = destination is FileStream or MemoryStream;
```

Comment:
Please change `var` to `bool` here.

Reply:
i am a fan of var, but i will change it to your style